src/main/java/org/truffleruby/language/CallStackManager.java

```diff
@@ -176,6 +176,11 @@ public Node getTopMostUserCallNode() {
         });
     }
 
+    @TruffleBoundary
+    public SourceSection getTopMostUserSourceSection() {
+        return getTopMostUserCallNode().getEncapsulatingSourceSection();
```

Comment:
What if `getTopMostUserCallNode()` returns `null`? I think `getTopMostUserSourceSection` should just return `null` as well in that case.

Reply:
Fixed, thanks.